front/src/modules/timesStops/helpers/utils.ts

```diff
@@ -125,3 +139,145 @@ export function disabledTextColumn(
     disabled: true,
   };
 }
+
+/**
+ * Synchronizes arrival, departure and stop times.
+ * updates onStopSignal
+ * updates isMarginValid and theoreticalMargin
+ */
+export function updateRowTimesAndMargin(
```

Comment:
this needs to be discussed with @axrolld 
a departure time doesn’t makes sense on its own, does it ?

Reply:
Could we consider that a departure time on its own is equivalent to a passing time ? I'm saying this because a use case i see would be if the user wants to set a stop, but he/she starts with the departure time input, then a stop time for example. In my opinion the departure time should be displayed if it is set by the user, and used either as a passing time if it's on its own, of used normally if it comes with an arrival time or a stop time

Reply:
If we don't do this, setting a stop becomes dependant on the right order of the cells filling, which to me isn't great from a UX perspective

Reply:
Ok maybe we can create an issue for this usecase to be solved as it doesn't really concern this PR

Reply:
we created this ticket https://github.com/OpenRailAssociation/osrd/issues/8894